### ISSUE TITLE
[Feature request] Allow `tf.estimator.train_and_evaluate` to evaluate on multiple datasets

### ISSUE THREAD

**Issue by joe-antognini** [tensorflow/tensorflow#20464]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No.
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: Binary
- **TensorFlow version (use command below)**: 1.8.0
- **Python version**: 2.7.12
- **Bazel version (if compiling from source)**: n/a
- **GCC/Compiler version (if compiling from source)**: n/a
- **CUDA/cuDNN version**: n/a
- **GPU model and memory**: n/a
- **Exact command to reproduce**: n/a

### Describe the problem

Currently `tf.estimator.train_and_evaluate` makes it easy to use an `Estimator` to perform both training and evaluation, possibly in a distributed environment.  However, this function only supports a single evaluation dataset.  This makes the function suboptimal because we oftentimes want to evaluate on both the training and the validation set in order to get a sense for the amount of overfitting that is happening.  It would be ideal if we could perhaps pass a list of `EvalSpec` objects to `train_and_evaluate`.

**Comment by candyzone** [comment]
By the way, I ask another related question.
When I use API `train_and_evaluate `, TenorFlow make a single node (not part of training cluster) that named `Evaluator`, it monitor the checkpoint directory and evaluate.
For large model, parameters' size cannot load by single node, Is there any suggestion about distributed evaluate in TensorFlow? Can evaluator reuse training cluster?

**Comment by bhack** [comment]
Probably is it a duplicate of https://github.com/tensorflow/tensorflow/issues/16087

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 16 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by bhack** [comment]
@jart I suppose that you want to reassign this right?

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 44 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 59 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 74 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 104 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 119 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @jart: It has been 134 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by Syndrome777** [comment]
> By the way, I ask another related question.
> When I use API `train_and_evaluate `, TenorFlow make a single node (not part of training cluster) that named `Evaluator`, it monitor the checkpoint directory and evaluate.
> For large model, parameters' size cannot load by single node, Is there any suggestion about distributed evaluate in TensorFlow? Can evaluator reuse training cluster?

I have the same issue. Do you have any suggestion?

**Comment by rmothukuru** [comment]
@joe-antognini,
Sorry for the delayed response. Since we use [TF Keras](https://www.tensorflow.org/api_docs/python/tf/keras) predominantly in **`Tensorflow 2.x`** and doesn't use [TF Estimators](https://www.tensorflow.org/guide/estimator) much, can you please let us know if this feature request is still relevant? Thanks!

**Comment by joe-antognini** [comment]
Yeah, I think that this isn't needed anymore since Estimators discouraged.
